src/gui_common/charts/line/LineChart.cs

```diff
@@ -0,0 +1,591 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Godot;
+using Array = Godot.Collections.Array;
+
+/// <summary>
+///   A custom widget for multi-line chart with hoverable data points tooltip. Uses <see cref="LineChartData"/>
+///   as dataset; currently only support numerical datas.
+/// </summary>
+public class LineChart : VBoxContainer
+{
+    [Export]
+    public NodePath HorizontalLabelPath;
+
+    [Export]
+    public NodePath VerticalLabelPath;
+
+    [Export]
+    public NodePath VerticalTicksContainerPath;
+
+    [Export]
+    public NodePath HorizontalTicksContainerPath;
+
+    [Export]
+    public NodePath DrawAreaPath;
+
+    [Export]
+    public NodePath LegendsContainerPath;
+
+    /// <summary>
+    ///   Number of scales to represent x axis values
+    /// </summary>
+    [Export]
+    public int XAxisTicks;
+
+    /// <summary>
+    ///   Number of scales to represent y axis values
+    /// </summary>
+    [Export]
+    public int YAxisTicks;
+
+    [Export]
+    public LegendDisplayMode LegendMode = LegendDisplayMode.Icon;
+
+    /// <summary>
+    ///   Limits how many icon legend shown
+    /// </summary>
+    public int MaxIconLegend = 10;
+
+    /// <summary>
+    ///   Limits how many dataset lines should be allowed to be shown on the chart.
+    /// </summary>
+    public int MaxDisplayedDataSet = 3;
+
+    /// <summary>
+    ///   Fallback icon for the legend display mode using icons
+    /// </summary>
+    private Texture defaultIconLegendTexture;
+
+    private Texture hLineTexture;
+
+    // ReSharper disable once NotAccessedField.Local
+    private Texture vLineTexture;
+
+    private Label horizontalLabel;
+    private Label verticalLabel;
+    private VBoxContainer verticalLabelsContainer;
+    private HBoxContainer horizontalLabelsContainer;
+    private Control drawArea;
+    private HBoxContainer legendContainer;
+
+    private string xAxisName;
+    private string yAxisName;
+
+    private List<ToolTipCallbackData> toolTipCallbacks = new List<ToolTipCallbackData>();
+
+    /// <summary>
+    ///   Modes on how the chart legend should be displayed
+    /// </summary>
+    public enum LegendDisplayMode
+    {
+        /// <summary>
+        ///   Legend will be displayed as rows of toggleable icons
+        /// </summary>
+        Icon,
+
+        /// <summary>
+        ///   Legend will be displayed as a dropdown button with a list of toggleable items
+        /// </summary>
+        DropDown,
+    }
+
+    // ReSharper disable once CollectionNeverUpdated.Global
+    // ReSharper disable RedundantNameQualifier
```

Comment:
This check shouldn't be worked around, instead specify a name alias at the top of the file.